Making manager more mobile friendly

jpdevries opened the pull request:
### What does it do ?
- Gets the fixed and absolute positioned silliness out of the way on
  small screens by combatting ExtJS' nature of aggressively preventing usability on small viewports
- fights :fire: with :fire: 
- Makes the left hand tree more responsive (can expand and collapse with a button at the bottom of the page)
- Make the resource edit screen mobile friendly
### Why is it needed ?

So it is more feasible to navigate the manager and edit resources on mobile devices.
### Related issue(s)/PR(s)

n/a
### Before

![](http://j4p.us/image/3W3V2M0d1h2b/Screen%20Shot%202015-11-21%20at%208.38.57%20PM.png)
### After

![](http://j4p.us/image/1w3D401U1y0X/Screen%20Shot%202015-11-21%20at%208.34.26%20PM.png)

OptimusCrime commented:
Pretty sure you should not include the rendered .css file. Other than that, awesome work!

pixelchutes commented:
This looks awesome, JP!

pixelchutes commented on manager/assets/modext/core/modx.component.js:
```diff
@@ -7,8 +7,9 @@ MODx.Component = function(config) {
     }
-    this._loadComponents();
     this._loadActionButtons();
+	console.log(document.getElementById('modx-container').innerHTML);
```
Any reason this shouldn't be removed?

jpdevries replied:
Lol whoops! That's what I get for submitting a pr during a MODXpo lecture. I'll fix it. 



—
Sent from Mailbox

On Sat, Nov 21, 2015 at 11:20 PM, Mike Reid <notifications@github.com>
wrote:

>>      this._loadActionButtons();
>> +	console.log(document.getElementById('modx-container').innerHTML);
> Any reason this shouldn't be removed?
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/modxcms/revolution/pull/12776/files#r45551647

jpdevries commented:
@OptimusCrime AFAIK it is part of the workflow to build the files before committing. When merges happen with CSS Integrators can just re-run `grunt build` to solve them.

I removed the console.log statement and cleaned up by squashing the commits into one

Mark-H commented:
@jpdevries Actually @OptimusCrime is right - to prevent conflicts from occurring it's better if the compiled CSS is not included in pull requests, the integrators will take care of updating that as things get merged. It's okay though, can sort it out either way. 

Could you update your PR with the template from https://github.com/modxcms/revolution/blob/2.x/CONTRIBUTING.md#template-1? Gotta set the right example ;)

jpdevries commented:
@Mark-H Oh I see, wasn't aware of that. done and done, i pulled the css out too

Mark-H commented:
Thanks man :) This looks really good from your screenshot, but it looks like it might need some further tweaks. Let's look at this together tomorrow? 

<img width="492" alt="screen shot 2015-11-22 at 01 08 48" src="https://cloud.githubusercontent.com/assets/312944/11321207/adb745e6-90b5-11e5-9474-14c77abbe80f.png">

jpdevries commented:
Huh, ya i am not sure what is going on there. I originally branched off develop, but submitted the pr to 2.4.x. Not sure if that could have anything to do with it or not. 



—
Sent from Mailbox

On Sun, Nov 22, 2015 at 1:09 AM, Mark Hamstra <notifications@github.com>
wrote:

> Thanks man :) This looks really good from your screenshot, but it looks like it might need some further tweaks. Let's look at this together tomorrow? 
> <img width="492" alt="screen shot 2015-11-22 at 01 08 48" src="https://cloud.githubusercontent.com/assets/312944/11321207/adb745e6-90b5-11e5-9474-14c77abbe80f.png">
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/modxcms/revolution/pull/12776#issuecomment-158692837

Mark-H commented:
That could be related for sure, let's find out tomorrow, could also be an issue with my dev environment as I just started a fresh one.

jpdevries commented:
I reproduced it kind of but then after I cleared cache and refreshed the page it looks right again. Ya let's look at it tomorrow. It would be nice to try and figure out a way to format the uber bar too and maybe even re order the preview button (using flex-order) so it is last and big like the help button is now. 



—
Sent from Mailbox

On Sun, Nov 22, 2015 at 1:19 AM, Mark Hamstra <notifications@github.com>
wrote:

> That could be related for sure, let's find out tomorrow, could also be an issue with my dev environment as I just started a fresh one. 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/modxcms/revolution/pull/12776#issuecomment-158693678

mindeffects commented:
Great idea for a PR, @jpdevries!
But why not kill those vertical lines in the menu bar at all? They are not needed for anything and are already gone on the right side of the manu bar.

jpdevries committed:
Making header more mobile friendly

Gets the fixed and absolute positioned silliness out of the way on
small screens

jpdevries commented:
@mindeffects i'm not opposed to removing the lines, just mostly been focusing on responsive layout with this PR.

@Mark-H I think I know what is going on. The layout is still messed up if the tree is open

### Tree Open :cry: 
![](http://j4p.us/image/1V1d233D1T2K/Screen%20Shot%202015-11-22%20at%2010.33.08%20AM.png)

### Tree Closed :smile: 
![](http://j4p.us/image/2T3Z1403160Y/Screen%20Shot%202015-11-22%20at%2010.32.40%20AM.png)

jpdevries commented:
With a bit more work I managed to get the lefthand sidebar (tree) to be more supported on mobile.

![](http://j4p.us/image/2B0v3x0W3m0P/Screen%20Shot%202015-11-22%20at%2012.00.27%20PM.png)

It is kind of weird how you expand and collapse the tree, but I'm making do with what I can. If you scroll down to the bottom of the page there is a button to expand / collapse the tree

![](http://j4p.us/image/0G2s381E310Z/Screen%20Shot%202015-11-22%20at%2012.02.07%20PM.png)

Haven't pushed yet but will soon.

jpdevries committed:
Making lefthand sidebar (tree) more mobile friendly

exside commented:
:+1:

jpdevries commented:
I intended to only work on the header (hints the responsive-header branch name) but figured I'd try and work my way down the page. How about making the content entry area a bit more mobile friendly?

![](http://j4p.us/image/0x1c0Z1T1q2c/Screen%20Shot%202015-11-22%20at%2012.18.06%20PM.png)

mindeffects commented:
Cool!

jpdevries committed:
Make Manager Edit page more mobile friendly

woohoo
Content tab
Settings tab

jpdevries commented:
Just updated the expand / collapse sidebar button to be responsive. 

It’s a little weird because the component is all the way at the bottom
of the page. Ideally it would be just beneath the tree sidebar itself,
but we haven’t been able to figure out how to change where the
component is ordered within the markup so this is the best we can do
for now

![](http://j4p.us/image/3R0g1O100n0J/Screen%20Shot%202015-11-22%20at%202.20.33%20PM.png)

Mark-H commented on manager/assets/modext/core/modx.component.js:
```diff
@@ -104,7 +104,7 @@ Ext.extend(MODx.Component,Ext.Component,{
         return true;
     }
 });
-Ext.reg('modx-component',MODx.Component);
```
Is there any particular reason you moved this line down to line 359?

jpdevries replied:
that is from when I was trying to figure out how to change the order of where the action buttons where on the page, but I do't think all those changes to modx.component.js are needed. I'll try removing some of them

jpdevries commented:
This PR does change where the MODX action bars get rendered to (so they can be before the main area) so I suppose that could cause some edge cases but all it is really doing is giving it another wrapper. If people are selecting the actions box by the id to do things like add their own buttons all that should work fine. 
Either way, I suppose targeting 2.5 is good to be safe.

jpdevries committed:
Responsive collapse / expand sidebar

It’s a little weird because the component is all the way at the bottom
of the page. Ideally it would be just beneath the tree sidebar itself,
but we haven’t been able to figure out how to change where the
component is ordered within the markup so this is the best we can do
for now

jpdevries commented:
I dld a little more work while I was :airplane: across the pond :smiley: 

### Uberbar Open
![screen shot 2015-11-23 at 7 05 16 pm](http://j4p.us/image/3W2F2W1y2C1e/Screen%20Shot%202015-11-23%20at%204.35.05%20PM.png)

### Create/Edit Template Variable
![screen shot 2015-11-23 at 2 08 41 pm](https://cloud.githubusercontent.com/assets/592335/11351965/3f2c644e-91ff-11e5-942f-e2e5a7b1efcd.png)

### Create/Edit Template
![screen shot 2015-11-23 at 2 08 27 pm](https://cloud.githubusercontent.com/assets/592335/11351958/34470cdc-91ff-11e5-836b-a9a2eba5c39b.png)

### Create/Edit Chunk
![screen shot 2015-11-23 at 2 09 06 pm](https://cloud.githubusercontent.com/assets/592335/11351977/4b022682-91ff-11e5-99d3-d1720f669490.png)

### Properties Tab
![screen shot 2015-11-23 at 3 22 32 pm](https://cloud.githubusercontent.com/assets/592335/11351991/5b8830aa-91ff-11e5-9296-d727c0632e9c.png)

### Create New Snippet
![screen shot 2015-11-23 at 3 25 04 pm](https://cloud.githubusercontent.com/assets/592335/11351993/668c693a-91ff-11e5-8bdb-26f8d3f71c20.png)

### Create New Plugin
![screen shot 2015-11-23 at 3 26 07 pm](https://cloud.githubusercontent.com/assets/592335/11351996/6e1d1b5e-91ff-11e5-9b0b-8a950a3a0364.png)

### New Category Modal
![screen shot 2015-11-23 at 3 37 54 pm](https://cloud.githubusercontent.com/assets/592335/11352008/79dde414-91ff-11e5-978d-0d344547476b.png)

### Import HTML
![screen shot 2015-11-23 at 4 00 46 pm](https://cloud.githubusercontent.com/assets/592335/11352013/85f5ce2e-91ff-11e5-82e9-b6e2190660b5.png)

### Media Sources
![screen shot 2015-11-23 at 4 34 07 pm](https://cloud.githubusercontent.com/assets/592335/11352019/8d7a66dc-91ff-11e5-8631-51912e083bb6.png)

### Package Management
![screen shot 2015-11-23 at 4 45 28 pm](https://cloud.githubusercontent.com/assets/592335/11352024/982c3e7a-91ff-11e5-96fb-acf4dbe7b047.png)

### Clear Console
![screen shot 2015-11-23 at 5 02 48 pm](https://cloud.githubusercontent.com/assets/592335/11352039/a6c5f246-91ff-11e5-908f-d7636a73cbfc.png)

### Settings Page
![screen shot 2015-11-23 at 7 09 31 pm](https://cloud.githubusercontent.com/assets/592335/11352054/b8fe76e0-91ff-11e5-9f44-f1121f21011e.png)

### Context Page
![screen shot 2015-11-23 at 7 25 42 pm](https://cloud.githubusercontent.com/assets/592335/11352061/c3af0b36-91ff-11e5-9c23-41a41d81d1b6.png)

### ACLs Page
![screen shot 2015-11-23 at 7 46 29 pm](https://cloud.githubusercontent.com/assets/592335/11352065/cb77b5b6-91ff-11e5-944a-e87d8ce8048e.png)

### Create Property Set
![screen shot 2015-11-23 at 7 51 35 pm](https://cloud.githubusercontent.com/assets/592335/11352077/d5a2a082-91ff-11e5-93db-155e5d5f1e5b.png)

### Lexicons Page
![screen shot 2015-11-23 at 7 52 03 pm](https://cloud.githubusercontent.com/assets/592335/11352080/e104e994-91ff-11e5-9c39-9e0b05397f8e.png)

### Namespaces Page
![screen shot 2015-11-23 at 7 53 12 pm](https://cloud.githubusercontent.com/assets/592335/11352084/ebf74acc-91ff-11e5-980a-8ab917ea3542.png)

hansek commented:
Good job :+1:

christianseel commented:
Fantastic job JP! :tada: :+1:

mindeffects commented:
Awesome!
I hope this PR makes it into 2.4.3!

jpdevries commented:
> I hope this PR makes it into 2.4.3!

@mindeffects I did as well but it has been slated for 2.5.0 mostly because it moves where the actions toolbar is in the DOM. It still uses the same exact id, so I see no reason why this would break anything. 

I intend to install 2.4.3 built from my fork so I can start my clients on it early. I can share the zip ;)

eighthday commented:
amazing! never thought this would come before modx 3

jaygilmore commented:
Squee!

rtripault commented:
Absolutely awesome work & effort! Huge :+1: from here!

My only concern is about code comments (as in many other core code places, unfortunately).
SASS allows comments, which could be "removed" when compiling, so i would love having loads of those.

It's hard for other maintainers/contributors to understand the intention, or even what's supposed to be targeted, without comments.

jpdevries commented:
@rtripault glad you like it!

The Sass is certainly nasty. Overly specific selectors and !important declarations are needed to override the inline styling of ExtJS. I could certainly add some comments to help organize things. Also, I would like to adjust the styles for certain components but would need some help from an ExtJS contributor that could help add a few classes in certain areas and maybe change the order of a couple of things in the DOM if possible.

pixelchutes commented:
@jpdevries I haven't had a chance to test this yet, but one observation on the "Uberbar Focused" and "Uberbar Results" images above:

It's seems to have regressed the 'up arrow' visual connector, which previously was designed to align with the magnifying glass at each width / breakpoint:

### Before
*Notice at the end (once resized smaller)*
![searchbar](https://cloud.githubusercontent.com/assets/352182/4196978/9fd8b65a-37e0-11e4-93fc-cd2198b13701.gif)

HarveyEV commented:
Awesomesauce! Can't wait for 2.5. :thumbsup:

jpdevries commented:
@pixelchutes good call on the arrow that should be an easy fix
![](http://j4p.us/image/2a0m1e3v1k1D/Screen%20Shot%202015-11-28%20at%2010.56.08%20AM.png)

jpdevries committed:
more mobile enhancements (momo)

Import HTML
Import Static HTML
Resource Groups
Content Types
Media Sources
Installer
Manager Customizations
Dashboard
Contexts
Menus
ACLs
Property Sets
Lexicon Management
Namespace

jpdevries commented:
updated
![](http://j4p.us/image/1y0Y1Z1G3H3y/Screen%20Shot%202015-11-28%20at%202.14.33%20PM.png)

Mark-H commented:
[This has been merged into 2.x](https://github.com/modxcms/revolution/commit/3824267f35ae40dd07260f33c6cf67c7e8598a76) for inclusion in 2.5, thank you @jpdevries for making MODX more usable!!

jpdevries commented:
@Mark-H what can I say i can't :zzz: on :airplane: 
Thanks for the merge and Happy December :smile:

noahlearner commented:
Will this work with touch?  A big part of how the manager works depends on right clicking menu items.  How will we do this on mobile device / tablet?

Mark-H commented:
@noahlearner This pull request doesn't change that, this just makes sure that you're actually able of navigating through the manager without wanting to poke your eyes out. :) Various mobile devices/tablets have different ways of doing right clicks.. long or force tabs typically. 

I'm sure there'll be further improvements in the future specifically to target the right clicks, but this one is not that yet.

noahlearner commented:
Will the solution look like a javascript solution to listen for a hold down to map to right click?

Kind of like the solution on: http://stackoverflow.com/questions/2625210/long-press-in-javascript

or the http://www.jquery-plugins.info/longclick-00014475.htm

rtripault commented:
Just rising an "slightly boring" issue : 

* change your browser viewport to a "mobile" one, ie. 615px (the tree is now at the top and its width is ~600px)
* restore to a "desktop width", notice the tree is still around 600px, with an horizontal scroollbar at the bottom
